dotnet/src/Experimental/Process.Core/ProcessBuilder.cs

```diff
@@ -96,31 +96,15 @@ internal override Dictionary<string, KernelFunctionMetadata> GetFunctionMetadata
     /// </summary>
     /// <param name="stateMetadata">State to apply to the step on the build process</param>
     /// <returns></returns>
-    internal override KernelProcessStepInfo BuildStep(KernelProcessStepStateMetadata<object>? stateMetadata)
+    internal override KernelProcessStepInfo BuildStep(KernelProcessStepStateMetadata? stateMetadata = null)
     {
         // The step is a, process so we can return the step info directly.
         if (stateMetadata is KernelProcessStateMetadata processState)
```

Comment:
Nit / Not Blocking - I'm wondering if there's ever a cae where `stateMetadata` might not be `KernelProcessStateMetadata`

Reply:
I guess now the base class will always be `KernelProcessStepStateMetadata` regardless if a new statemetadata gets added or not.